### ISSUE TITLE
allow long-press on boost to boost from other than currently active account

### ISSUE THREAD

**Issue by foss-** [pachli/pachli-android#189]
**Is your feature request related to a problem?**
When seeing a post that would be nice to be shared but shared in another than the account currently viewing the post, it is not easy to share that post via one of the other accounts used in Pachli.

**Describe the solution you'd like**
Fedilab has a nice feature where a long press on the boost icon shows a dialog `Chose an account` with a list of signed in accounts in Fedilab.

Certainly not a priority feature however a nice quality of life improvement.

**Comment by nikclayton** [comment]
Agreed. That's one of the reasons why the boost/favourite confirmation options show as a menu. In the future that menu will grow to include a "Boost as <other account>", and appear on long-press too.

It's more complicated than it should be at the moment because a lot of the "perform an action on a status" code assumes that the currently logged-in account is the one to use, and needs to be refactored to receive the account to perform the action with as a parameter.

I've labelled this as "good first issue" because although that's a reasonable amount of work it should be relatively straightforward. If anyone's reading this and wants to try follow up here and I'll jot down some more notes.
